pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go

```diff
@@ -79,23 +79,11 @@ func (pl *NodeAffinity) Score(ctx context.Context, state *framework.CycleState,
 	// An element of PreferredDuringSchedulingIgnoredDuringExecution that refers to an
 	// empty PreferredSchedulingTerm matches all objects.
 	if affinity != nil && affinity.NodeAffinity != nil && affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution != nil {
-		// Match PreferredDuringSchedulingIgnoredDuringExecution term by term.
-		for i := range affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution {
-			preferredSchedulingTerm := &affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution[i]
-			if preferredSchedulingTerm.Weight == 0 {
-				continue
-			}
-
-			// TODO: Avoid computing it for all nodes if this becomes a performance problem.
-			matches, err := corev1.MatchNodeSelectorTerms(node, &v1.NodeSelector{NodeSelectorTerms: []v1.NodeSelectorTerm{preferredSchedulingTerm.Preference}})
-			if err != nil {
-				return 0, framework.AsStatus(err)
-			}
-
-			if matches {
-				count += int64(preferredSchedulingTerm.Weight)
-			}
+		preferredNodeAffinity, err := nodeaffinity.NewPreferredSchedulingTerms(affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution)
```

Comment:
Does this still compute it for all nodes? Maybe I am missing where the computed affinity is stored, or that will be added in another PR, just curious

Reply:
Correct. For that, I would need to implement PreFilter, which is out of the scope for this PR. But I might leave that for 1.21. The priority is implementing #95738, for which parsing happens during plugin instantiation.

Reply:
Sounds good, just wanted to make sure I understood thanks

Reply:
I created an issue and restored the TODO